internal/dataplane/kong_client.go

```diff
@@ -511,13 +512,28 @@ func (c *KongClient) maybeSendOutToKonnectClient(ctx context.Context, s *kongsta
 			c.logger.Error(err, "Skipped pushing configuration to Konnect")
 		} else {
 			c.logger.Error(err, "Failed pushing configuration to Konnect")
+			logKonnectErrors(c.logger, err)
 		}
 		return err
 	}
 
 	return nil
 }
 
+// logKonnectErrors logs details of each error response returned from Konnect API.
+func logKonnectErrors(logger logr.Logger, err error) {
+	if apiErrors := deckerrors.ExtractAPIErrors(err); len(apiErrors) > 0 {
+		for _, apiErr := range apiErrors {
+			// REVIEW: log level? debug, info or error? Use debug here to prevent from too many logs?
+			logger.Info("Failed to send request to Konnect",
+				"code", apiErr.Code(),
+				// REVIEW: json.Marshal here?
```

Comment:
I think that we should leave it up to the logger. I've checked that with the logger we're using a structure is properly made a part of the JSON that's written to the buffer:

```go
func TestLoggerWithStructField(t *testing.T) {
	for _, format := range []string{"console", "json"} {
		zapLogger, err := util.MakeLogger("debug", format, os.Stderr)
		require.NoError(t, err)
		logger := zapr.NewLogger(zapLogger)

		s := struct {
			Name   string
			Number int
			Nested struct {
				Name string
			}
		}{
			Name:   "hello",
			Number: 1,
			Nested: struct {
				Name string
			}{
				Name: "world",
			},
		}

		logger.Info("hello world", "key", "value", "struct", s)
	}
}
```
Gives:
```console
2024-01-17T12:56:46+01:00	info	hello world	{"key": "value", "struct": {"Name":"hello","Number":1,"Nested":{"Name":"world"}}}
{"level":"info","time":"2024-01-17T12:56:46+01:00","msg":"hello world","key":"value","struct":{"Name":"hello","Number":1,"Nested":{"Name":"world"}}}
```

If we marshaled the structure ourselves, we'd make it harder to query it because it would be a string field in JSON, like below:

```console
2024-01-17T13:01:04+01:00	info	hello world	{"key": "value", "struct": "{\"Name\":\"hello\",\"Number\":1,\"Nested\":{\"Name\":\"world\"}}"}
{"level":"info","time":"2024-01-17T13:01:04+01:00","msg":"hello world","key":"value","struct":"{\"Name\":\"hello\",\"Number\":1,\"Nested\":{\"Name\":\"world\"}}"}
```

Reply:
Done.